devito/types/sparse.py

```diff
@@ -324,14 +323,13 @@ class AbstractSparseTimeFunction(AbstractSparseFunction):
     _time_position = 0
     """Position of time index among the function indices."""
 
-    def __init__(self, *args, **kwargs):
-        if not self._cached():
-            self._time_dim = self.indices[self._time_position]
-            self._time_order = kwargs.get('time_order', 1)
-            if not isinstance(self.time_order, int):
-                raise ValueError("`time_order` must be int")
+    def __init_finalize__(self, *args, **kwargs):
```

Comment:
note for reviewers: this is exactly like before, but de-indented as we don't need ``if not self._cached()`` anymore